draft-specifications/CapTP Specification.md

```diff
@@ -0,0 +1,629 @@
+
+This operation is used when a new connection is initiated over CapTP. Both parties MUST send upon a new connection. The operation looks like this:
+
+```text
+<op:start-session [captp-version              ; Floating point value
+                   session-pubkey             ; CapTP public key value
+                   acceptable-location        ; OCapN Reference type
+                   acceptable-location-sig]>  ; CapTP signature
+```
+
+It's important that we only have one bidirectional connection between a CapTP session. Before trying to connect to a machine, it's important to check that a connection is not already open before proceeding. There is a race condition called the "crossed hellos" problem whereby each side tries to open a new connection at the same time and only one of these connection attempts must succeed. The way to check which attempt should succeed is to sort the serialized `session-pubkey`, the highest of the two wins.
+
+### Constructing and sending
+
+The `captp-version` MUST be `1.0`.
+
+The `session-pubkey` is the public key part of the per-session key pair generated for this connection. This is serialized in accordance with [Cryptography](#cryptography)
+
+The `acceptable-location` is a OCapN Locator which represents the location where the session is accessable.
+
+The `acceptable-location-sig` is the signature of the serialized `acceptable-location`. The signature is created using the private key from the per-session key pair. This is serialized in accordance with [Cryptography](#cryptography)
+
+### Receiving
+
+The `captp-version` MUST be equal to `1.0`. If the version does not match, the connection MUST be aborted.
+
+The `acceptable-location-sig` MUST be valid that the `session-pubkey` provided a valid signature of `acceptable-location`.
+
+## [`op:bootstrap`](#op-bootstrap)
+
+This operation requests the bootstrap object to be available at the specified `answer-position`. The message looks like:
+
+```text
+<op:bootstrap [answer-position    ; positive integer or 0
+               resolve-me-desc]>  ; desc:import-object | desc:import-promise
+```
+
+### The bootstrap Object
+
+The bootstrap object is responsible for providing access to local objects on the session. It has two different behaviors, these are selected using the conventional CapTP method mechanism of sending a symbol as the first argument, the following methods are available:
+
+-   `fetch`
+-   `deposit-gift`
+-   `withdraw-gift`
+
+#### `fetch` Method
+
+This method is used to fetch an object from the bootstrap object. To use it you need a `swiss-number` which is a Binary Data type. This swiss number should correspond an object which exists in this session. The result will be the object which corresponds to this `swiss-number` or an error if the object does not exist or a swiss number was not provided.
+
+Here is an example of how to use this method is:
```

Comment:
Maybe either "Here is an example of how to use this method" or "An example of how to use this method is:"